editions/tw5.com/tiddlers/filters/syntax/Cascade Filter Run Prefix (Examples).tid

```diff
@@ -0,0 +1,22 @@
+created: 20211130114857532
+modified: 20211130121656441
+tags: [[Filter Run Prefix Examples]] [[Filter Syntax]] [[Cascade Filter Run Prefix]]
+title: Cascade Filter Run Prefix (Examples)
+type: text/vnd.tiddlywiki
+
+The <<.op ":cascade">> filter run prefix is used by the ~TiddlyWiki core to choose the icon for each tiddler.
+
+`:cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerColourFilter]get[text]]`
+
+The filter expression inside the above <<.op :cascade>> filter run returns a list of filters from the text fields of tiddlers tagged <<tag $:/tags/TiddlerColourFilter>> as shown below:
+
+<ul>
+<$list filter="[all[shadows+tiddlers]tag[$:/tags/TiddlerColourFilter]get[text]]">
+<li><pre><$text text=<<currentTiddler>>/></pre></li>
+</$list>
+</ul>
+
+For each input title, each of these filters is now applied to in turn until the first filter that returns a non-empty list. The first result from this non-empty list replaces that input title and is used as the icon for that tiddler.
+
+<<.operator-example 1 """[all[shadows+tiddlers]] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerColourFilter]get[text]] +[!is[blank]limit[10]]""">>
```

Comment:
I think this example doesn't make sense here. IMO it would be better to describe the 3 filters similar, to the [red text](at: https://github.com/Jermolene/TiddlyWiki5/discussions/6280#discussioncomment-1698503 ) I used at the GH discussion

There should also be some info about the ControlPanel : Info : Advanced : Tiddler Colours tab where the users can see the active cascades.

The text I used: 

1: If a tiddler has a "color" field it will be used as the "tag-pill" colour if the tiddler is a tag. If the currentTiddler has no color field the program goes to the next filter

2: If the $:/config/DefaultTiddlerColour tiddler exists, use the text as the default color. This allows us to globally change the default colour of the tag-pills without changing a core tiddler

3: The color attribute of all tiddlers tagged: "TableOfContent" will be blue-ish
The color attribute of all tiddlers tagged: "Working with TiddlyWiki" will be purpel-ish

Have a look at the icon in "GettingStarted" and the GettingStarted tag-pill